### PR TITLE
Algolia - update search index

### PR DESCRIPTION
**Pull request by alebabai**
## what
use an NPM package for running atomic updates to Algolia indices

## why
#144

### CODE REVIEW

**Comment by alebabai** [comment]
Should be merged after #154

**Comment by alebabai** [comment]
Requires following environment variables:

- **ALGOLIA_APP_ID**: the ID of the Algolia app instance that contains your index.
- **ALGOLIA_ADMIN_KEY**: the adminstrative key for the Algolia app instance that contains your index.
- **ALGOLIA_INDEX_NAME**: the name of the index you're updating.
- **ALGOLIA_INDEX_FILE**: the relative path to your index file from the root of your project.

**Comment by alebabai** [comment]
Should i use fake ones for `ALGOLIA_ADMIN_KEY`, `ALGOLIA_APP_ID`, etc?

**Comment by osterman** [comment]
> Should i use fake ones 

Yea, do that for now. PM me and I can send you one for testing.

**Comment by alebabai** [comment]
> How does atomic-algolia know to process index.algolia.json?
It seems to me that algolia index file uses some predefined standard - specified in list.algolia.json
https://forestry.io/blog/search-with-algolia-in-hugo/#creating-the-json-template

**Comment by alebabai** [comment]
I've updated environment variables in the CF, so it's ready to build.

File: Makefile [hunk 1]
```diff
@@ -15,9 +15,9 @@ export HTMLTEST_LOG_LEVEL ?= 2
 export HTMLTEST_CONFIG ?= .htmltest.yml
 
 export ALGOLIA_INDEX_FILE ?= $(HUGO_PUBLISH_DIR)/index.algolia.json
-export ALGOLIA_APPLICATION_INDEX ?= dev
-export ALGOLIA_API_ENDPOINT ?= "https://$(ALGOLIA_APPLICATION_ID).algolia.net/1/indexes/$(ALGOLIA_APPLICATION_INDEX)"
-#export ALGOLIA_API_ENDPOINT ?= "https://httpbin.org/post"
+export ALGOLIA_INDEX_NAME ?= dev
+export ALGOLIA_APP_ID ?= docs
+# export ALGOLIA_ADMIN_KEY =? <token>
```
Comment (osterman) [inline Makefile]: Change this to just:
```
export ALGOLIA_ADMIN_KEY =?
```

Comment (osterman) [reply Makefile]: Drop the comment

**Pull request merged**